express-notes-api/index.js

```diff
@@ -0,0 +1,97 @@
+const express = require('express');
+const app = express();
+const data = require('./data.json');
+
+app.use(express.json());
+
+// GET
+
+app.get('/api/notes', (req, res) => {
+  const arr = [];
+  const notes = data.notes;
+  for (const property in notes) {
+    arr.push(notes[property]);
+  }
+  res.send(arr);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const notes = data.notes;
+  if (isNaN(req.params.id) || req.params.id < 0) {
+    res.status(400).send({ error: 'id must be a positive integer' });
+  }
+  if (req.params.id === undefined) {
```

Comment:
This is fine but you can also just do:
```JAVASCRIPT
if (!req.params.id) {
```

You can do that whenever you are checking if a property exists or not